### PR TITLE
SeparableMetzArrayFilter uses stir::Fourier

### LINKED ISSUE #204: SeparableMetzArrayFilter move to stir::fourier
discrete_fourier_transform in SeparableMetzArrayFilter.cxx should be deleted and replace by stir::fourier. Should be trivial, aside of course from conventions on where the zero is etc etc…. would have to do some stand-alone checks I guess. Or see what the creation of a postfilter says, as it lists the coeffs

### PR DESCRIPTION
**Pull request by NikEfth**
closes #204

### CODE REVIEW

**Comment by NikEfth** [comment]
One travis build failed with : 
```
*** Break *** segmentation violation
[/usr/lib/system/libsystem_platform.dylib] _sigtramp (no debug info)
[<unknown binary>] (no debug info)
[/Users/travis/build/UCL/STIR/install/bin/lm_to_projdata] stir::CListEventCylindricalScannerWithDiscreteDetectors::get_bin(stir::Bin&, stir::ProjDataInfo const&) const (no debug info)
[/Users/travis/build/UCL/STIR/install/bin/lm_to_projdata] stir::LmToProjData::process_data() (no debug info)
[/Users/travis/build/UCL/STIR/install/bin/lm_to_projdata] main (no debug info)
[/usr/lib/system/libdyld.dylib] start (no debug info)
[<unknown binary>] (no debug info)
```

File: src/buildblock/SeparableMetzArrayFilter.cxx [hunk 1]
```diff
@@ -240,38 +184,29 @@ void build_metz(VectorWithOffset<elemT>& kernel,
     
     /* Build the fft array, odd coefficients are the imaginary part */
```
Comment (KrisThielemans) [inline src/buildblock/SeparableMetzArrayFilter.cxx]: change comment

File: src/buildblock/SeparableMetzArrayFilter.cxx [hunk 2]
```diff
@@ -227,9 +168,12 @@ void build_metz(VectorWithOffset<elemT>& kernel,
     
     //MJ 05/03/2000 padded 1 more element to fftdata and pre-increment
```
Comment (KrisThielemans) [inline src/buildblock/SeparableMetzArrayFilter.cxx]: remove comment

File: src/buildblock/SeparableMetzArrayFilter.cxx [hunk 3]
```diff
@@ -240,38 +184,29 @@ void build_metz(VectorWithOffset<elemT>& kernel,
...
-    
+
     /* FFT to frequency space */
-    fftdata.set_offset(1);
-    discrete_fourier_transform(fftdata/*-1*/,Res,FORWARDFFT); 
-    fftdata.set_offset(0);
-    
-    
-    
+    fourier< stir::Array<1,std::complex<elemT> > >(fftdata);
```
Comment (KrisThielemans) [inline src/buildblock/SeparableMetzArrayFilter.cxx]: do you actually need the explicit type? I didn't think so

File: src/buildblock/SeparableMetzArrayFilter.cxx [hunk 4]
```diff
@@ -282,36 +217,26 @@ void build_metz(VectorWithOffset<elemT>& kernel,
...
-    discrete_fourier_transform(fftdata/*-1*/,Res,INVERSEFFT); 
-    fftdata.set_offset(0);
-    
-    
-    
-    
-    
-    
+    /* return to the spatial space */
+    inverse_fourier< Array<1,std::complex<elemT> > >(fftdata);
```
Comment (KrisThielemans) [inline src/buildblock/SeparableMetzArrayFilter.cxx]: same about type

**Comment by KrisThielemans** [comment]
I've restarted the Travis build. it's worrying of course, but unrelated to this PR

File: src/buildblock/SeparableMetzArrayFilter.cxx (hunk 2 above)
Comment (NikEfth) [reply src/buildblock/SeparableMetzArrayFilter.cxx]: :ok:

File: src/buildblock/SeparableMetzArrayFilter.cxx (hunk 3 above)
Comment (NikEfth) [reply src/buildblock/SeparableMetzArrayFilter.cxx]: Indeed. I thought it was more readable. 
:ok:

File: src/buildblock/SeparableMetzArrayFilter.cxx (hunk 4 above)
Comment (NikEfth) [reply src/buildblock/SeparableMetzArrayFilter.cxx]: :ok:

**Comment by NikEfth** [comment]
I understand. The one before did not fail.

File: src/buildblock/SeparableMetzArrayFilter.cxx (hunk 1 above)
Comment (NikEfth) [reply src/buildblock/SeparableMetzArrayFilter.cxx]: :ok:

**Comment by KrisThielemans** [comment]
squash merge?

**Comment by NikEfth** [comment]
Alright

**Pull request merged**